src/Castle.Core.Tests/DynamicProxy.Tests/ParameterDefaultValuesTestCase.cs

```diff
@@ -0,0 +1,406 @@
+		[TestCase(nameof(HasDefaultValues.Long_nullable_non_default))]
+		[TestCase(nameof(HasDefaultValues.SByte_nullable_default))]
+		[TestCase(nameof(HasDefaultValues.SByte_nullable_non_default))]
+		[TestCase(nameof(HasDefaultValues.Short_nullable_default))]
+		[TestCase(nameof(HasDefaultValues.Short_nullable_non_default))]
+		[TestCase(nameof(HasDefaultValues.UInt_nullable_default))]
+		[TestCase(nameof(HasDefaultValues.UInt_nullable_non_default))]
+		[TestCase(nameof(HasDefaultValues.ULong_nullable_default))]
+		[TestCase(nameof(HasDefaultValues.ULong_nullable_non_default))]
+		[TestCase(nameof(HasDefaultValues.UShort_nullable_default))]
+		[TestCase(nameof(HasDefaultValues.UShort_nullable_non_default))]
+		public void Not_supported_on_Mono_Nullable_non_null(string methodName)
+		{
+			AssertParameter(typeof(HasDefaultValues), methodName);
+		}
+
+		[ExcludeOnClr("ParameterBuilder.SetConstant does not accept non-null default values for nullable enum parameters. See https://github.com/dotnet/coreclr/issues/17893.")]
+		[ExcludeOnMono("ParameterBuilder.SetConstant does not accept non-null default values for nullable parameters. See https://github.com/mono/mono/issues/8597.")]
+		[TestCase(nameof(HasDefaultValues.UserDefinedEnum_nullable_default))]
+		[TestCase(nameof(HasDefaultValues.UserDefinedEnum_nullable_non_default))]
+		public void Not_supported_UserDefinedEnum_nullable_non_null(string methodName)
+		{
+			AssertParameter(typeof(HasDefaultValues), methodName);
+		}
+
+		private void AssertParameter(Type type, string methodName)
+		{
+			var method = type.GetMethod(methodName);
+			var expectedDefaultValue = method.Invoke(Activator.CreateInstance(type), new object[1]);
+
+			var proxiedParameter = GetProxiedParameter(type, methodName);
+#if DOTNET45
+			Assert.True(proxiedParameter.HasDefaultValue);
+#endif
+			Assert.AreEqual(expectedDefaultValue, proxiedParameter.DefaultValue);
+		}
+
+		private static ParameterInfo GetOriginalParameter(Type type, string methodName = "Method")
+		{
+			return type.GetMethod(methodName).GetParameters()[0];
+		}
+
+		private ParameterInfo GetProxiedParameter(Type type, string methodName = "Method")
+		{
+			var proxy = generator.CreateClassProxy(type, new DoNothingInterceptor());
+			return proxy.GetType().GetMethod(methodName).GetParameters()[0];
+		}
+
+		public class HasDefaultValues
+		{
+			// This class has methods with optional parameters having all kinds of default values.
+			//
+			// Method name nomenclature used: `{parameter type}[_nullable]_{kind of default value}`
+			//
+			// The methods must return the default value that reflection is expected to report for the
+			// parameter. Typically this matches the declared default value, but there are exceptions!
+			//
+			// This method list is intended to be exhaustive, i.e. it should cover *all* possible
+			// kinds of default values. If you find a case that is missing, please add it. (Some lines
```

Comment:
I'd pull the sentence in parenthesis beginning "Some lines" into its own paragraph so it is more obvious.